src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -40,7 +46,41 @@
      * София - 900</p>
      */
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
-            throws Exception {
-        return null;
+            throws IllegalDateParametersException {
+
+        if (!compareDates(dateFrom,dateTo)) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+
+        StringBuilder sb = new StringBuilder("Отчёт за период " + dateFrom + " - " + dateTo + '\n');
+        ArrayList<String[]> dataToArray = new ArrayList<>();
+
+        for (String s : data) {
+            if (compareDates(dateFrom,s.substring(0,10))
+                    && compareDates(s.substring(0,10),dateTo)) {
+                dataToArray.add(s.split(" "));
+            }
+        }
+
+        int[] money = new int[names.length];
+        for (int i = 0; i < names.length; i++) {
+            for (String[] dataArray : dataToArray) {
+                if (names[i].equals(dataArray[1])) {
+                    money[i] += Integer.parseInt(dataArray[2]) * Integer.parseInt(dataArray[3]);
+                }
+            }
+        }
+
+        for (int i = 0; i < names.length; i++) {
+            sb.append(names[i]).append(" - ").append(money[i]).append('\n');
+        }
+        return sb.deleteCharAt(sb.length() - 1).toString();
+    }
+
+    public boolean compareDates(String d1, String d2) {
```

Comment:
Imagine I have to use this method, But i Can't see it's implementation. How should I know that you are using isBefore inside but nor isAfter for example?